python/cuml/cuml/accel/estimator_proxy.py

```diff
@@ -14,354 +14,110 @@
 # limitations under the License.
 #
 
+import importlib
 import inspect
-import sys
-import types
-from typing import Any, Dict, List, Optional, Tuple, Type
 
-from cuml.internals import logger
-from cuml.internals.global_settings import GlobalSettings
-from cuml.internals.safe_imports import cpu_only_import, gpu_only_import
 
-
-class ProxyModule:
-    """
-    A proxy module that dynamically replaces specified classes with proxy estimators
-    based on GlobalSettings.
-
-    Parameters
-    ----------
-    original_module : module
-        The module to be proxied.
-    Attributes
-    ----------
-    _original_module : module
-        The original module being proxied.
-    _proxy_estimators : dict of str to type
-        A dictionary mapping accelerated class names to their proxy estimators.
-    """
-
-    def __init__(self, original_module: types.ModuleType) -> None:
-        """Initialize the ProxyModule with the original module."""
-        self._original_module = original_module
-        self._proxy_estimators: Dict[str, Type[Any]] = {}
-
-    def add_estimator(
-        self, class_name: str, proxy_estimator: Type[Any]
-    ) -> None:
+def reconstruct_proxy(proxy_module, proxy_name, state):
+    module = importlib.import_module(proxy_module)
+    cls = getattr(module, proxy_name)
+    obj = cls.__new__(cls)
+    obj.__setstate__(state)
+    return obj
+
+
+class ProxyMixin:
```

Comment:
Follow up from my question before, this mixin will go away when the new proxyEstimator exists, right?

Reply:
Yes